testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/propagation/OidcIdentityPropagationTestCase.java

```diff
@@ -0,0 +1,646 @@
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.testcontainers.DockerClientFactory;
+import org.wildfly.extension.elytron.oidc.ElytronOidcExtension;
+import org.wildfly.security.permission.ElytronPermission;
+import org.wildfly.test.integration.elytron.oidc.client.KeycloakConfiguration;
+import org.wildfly.test.integration.elytron.oidc.client.OidcBaseTest;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.base.WhoAmIBean;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.ComplexServletLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.EntryBeanLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.EntryLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.ManagementBeanLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.ServletOnlyLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.SimpleServletLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.WhoAmIBeanLocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.local.WhoAmILocal;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.ComplexServletRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.EntryBeanRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.EntryRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.ManagementBeanRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.ServletOnlyRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.SimpleServletRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.WhoAmIBeanRemote;
+import org.wildfly.test.integration.elytron.oidc.client.propagation.remote.WhoAmIRemote;
+import org.wildfly.test.security.common.elytron.EjbElytronDomainSetup;
+import org.wildfly.test.security.common.elytron.ElytronDomainSetup;
+
+import io.restassured.RestAssured;
+
+/**
+ * Test class to hold the identity propagation scenarios involving a virtual security domain with OpenID Connect.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ OidcIdentityPropagationTestCase.EJBDomainSetupOverride.class, OidcIdentityPropagationTestCase.PropagationSetup.class, OidcIdentityPropagationTestCase.KeycloakAndSubsystemSetup.class })
+public class OidcIdentityPropagationTestCase {
```

Comment:
I suggest to add/update
- negative cases
  - where the outflow/trust is not configured on one of the sides
  - where the outflowed identity is not available in the trusting security domain
- some current positive cases with `outflow-anonymous` enabled + test(s) for outflow not possible resulting in anonymous identity used

Reply:
I've added additional tests cases for this.

Reply:
This was just meant to mimic the existing `outflow-anonymous` attribute for existing `security-domains`. If the target security domain already had an identity established, `outflow-anonymous` can result in the identity for that target security domain being cleared. It's described in the docs like this:

`Outflowing anonymous has the effect of clearing any identity already established for that domain.`

I found it a bit tricky to simulate this with the way the tests are set up but will try to take a look.

Reply:
Ah, yes, I'm sorry I was too fast and forgot about that _clearing_ part, thank you.